src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets

```diff
@@ -65,7 +65,7 @@
   <!-- Unless this is an inner build or default timing is disabled, tie document retrieval into the build. -->
 
   <Target Name="_GenerateOpenApiDocuments"
-      AfterTargets="Build"
+      AfterTargets="CoreBuild"
```

Comment:
@rainersigwald is it expected that `<MSBuild/>` tasks calling the `Build` target can return to the project (and continue) before the referenced project does `AfterBuild="Build"` work?

Reply:
Kind of! The semantics of "what does it mean to hook onto a project's entry-point target with AfterBuild" are very muddy.

Were you seeing execution return, or was it just continuing past an error in AfterTargets="Build"? If so, that's microsoft/msbuild#3345.

Reply:
The occasional failures looked like a race condition between referenced projects building (including document generation) and the client project starting on its `BeforeTargets="BeforeCompile"` work (getting filenames of generated documents). To see the output I recall, the server project must have been running two targets simultaneously -- `GenerateOpenApiDocuments` and `OpenApiGetDocuments`. In particular, I saw a [`<Message/>`](https://github.com/aspnet/AspNetCore/pull/13364/files#diff-962355bc963c7ef291c534e32b08cb77R61) about tool execution just before or just after an [`<Error/>`](https://github.com/aspnet/AspNetCore/pull/13364/files#diff-962355bc963c7ef291c534e32b08cb77R28) about a missing file that the command writes.

Put another way, I was seeing the occasional "client" (project with an `@(OpenApiProjectReference)` item) build continue before the "server" (referenced project using this code) build completed. No errors until the client called the `GetOpenApiDocuments` target when the server had not finished running `GenerateOpenApiDocuments`.

Reply:
Interesting. I think something deeper is wrong in this case. Any chance you have an MSBuild binlog of a build like this? Even a build that didn't fail due to the race would be instructive.

Reply:
I'll get something to you early next week